metadata-ingestion/src/datahub/ingestion/source/tableau.py

```diff
@@ -1372,19 +1624,23 @@ def emit_sheets_as_charts(
 
         if (
             workbook is not None
-            and workbook.get("projectName")
+            and workbook.get("projectLuid")
+            and workbook["projectLuid"] in self.tableau_project_registry
             and workbook.get("name")
         ):
-            # Browse path
-            browse_path = BrowsePathsClass(
+
+            browse_paths = BrowsePathsClass(
                 paths=[
-                    f"/{self.platform}/{workbook['projectName'].replace('/', REPLACE_SLASH_CHAR)}"
-                    f"/{workbook['name']}"
+                    f"/{self.platform}/{self._get_project_browse_path(workbook['projectLuid'])}"
```

Comment:
this is being made safer?

Reply:
None check is done at line number 1629 and 1630